interfaces/apparmor/backend_test.go

```diff
@@ -191,6 +191,174 @@ func (s *backendSuite) TestName(c *C) {
 	c.Check(s.Backend.Name(), Equals, interfaces.SecurityAppArmor)
 }
 
+type expSnapConfineTransitionRules struct {
+	usrBinSnapRules   bool
+	usrLibSnapdTarget string
+	coreSnapTarget    string
+	snapdSnapTarget   string
+}
+
+func checkProfileExtraRules(c *C, profile string, exp expSnapConfineTransitionRules) {
+	if exp.usrBinSnapRules {
+		c.Assert(profile, testutil.FileContains, "  /usr/bin/snap ixr,")
+		c.Assert(profile, testutil.FileContains, " /snap/{snapd,core}/*/usr/bin/snap ixr,")
+	} else {
+		c.Assert(profile, Not(testutil.FileContains), "  /usr/bin/snap ixr,")
+		c.Assert(profile, Not(testutil.FileContains), " /snap/{snapd,core}/*/usr/bin/snap ixr,")
+	}
+
+	if exp.usrLibSnapdTarget != "" {
+		rule := fmt.Sprintf("/usr/lib/snapd/snap-confine Pxr -> %s,", exp.usrLibSnapdTarget)
+		c.Assert(profile, testutil.FileContains, rule)
+	} else {
+		c.Assert(profile, Not(testutil.FileMatches), "/usr/lib/snapd/snap-confine Pxr ->.*,")
+	}
+
+	if exp.coreSnapTarget != "" {
+		rule := fmt.Sprintf("/snap/core/*/usr/lib/snapd/snap-confine Pxr -> %s,", exp.coreSnapTarget)
+		c.Assert(profile, testutil.FileContains, rule)
+	} else {
+		c.Assert(profile, Not(testutil.FileMatches), "/snap/core/*/usr/lib/snapd/snap-confine Pxr ->.*,")
```

Comment:
The pattern is a regex, so we probably want to have `/snap/core/.*/` (zero or more of any char) rather than `/snap/core/*/` (zero or more of /)
```suggestion
		c.Assert(profile, Not(testutil.FileMatches), "/snap/core/.*/usr/lib/snapd/snap-confine")
```